### PR TITLE
add storage extension

### PR DESCRIPTION
**Pull request by geospatial-jeff**
Move storage extension as it existed in https://github.com/radiantearth/stac-spec/pull/991

cc @davidraleigh

### CODE REVIEW

File: examples/example-naip.json [hunk 1]
```diff
@@ -0,0 +1,144 @@
+{
+  "stac_version": "1.0.0-beta.2",
```
Comment (philvarner) [inline examples/example-naip.json:2]: uses of beta.2 should be updated to either 1.0.0 or 1.0.0-rc.1

File: json-schema/schema.json [hunk 2]
```diff
@@ -1,138 +1,102 @@
...
-          "$ref": "#/definitions/stac_extensions"
-        },
-        {
-          "$ref": "#/definitions/fields"
-        }
-      ]
-    }
-  ],
+  "$id": "https://stac-extensions.github.io/storage/v1.0.0/schema.json",
+  "title": "Storage Object Extension",
```
Comment (philvarner) [inline json-schema/schema.json:4]:
```suggestion
  "title": "Storage Extension",
```

File: examples/example-naip.json (hunk 1 above)
Comment (philvarner) [inline examples/example-naip.json:2]:
```suggestion
  "stac_version": "1.0.0",
```

File: examples/example-nsl.json [hunk 3]
```diff
@@ -0,0 +1,121 @@
+{
+  "stac_version": "1.0.0-beta.2",
```
Comment (philvarner) [inline examples/example-nsl.json:2]:
```suggestion
  "stac_version": "1.0.0",
```

File: json-schema/schema.json [hunk 4]
```diff
@@ -1,138 +1,102 @@
...
+  "description": "STAC Storage Object Extension to a STAC Item and STAC Assets.",
```
Comment (philvarner) [inline json-schema/schema.json:5]:
```suggestion
  "description": "STAC Storage Extension to a STAC Item and STAC Assets.",
```

File: json-schema/schema.json [hunk 5]
```diff
@@ -1,138 +1,102 @@
...
+        "stac_extensions",
+        "properties",
+        "assets"
       ],
       "properties": {
         "stac_extensions": {
           "type": "array",
           "contains": {
-            "const": "https://stac-extensions.github.io/template/v1.0.0/schema.json"
+            "const": "https://stac-extensions.github.io/sar/v1.0.0/schema.json"
```
Comment (philvarner) [inline json-schema/schema.json:18]:
```suggestion
            "const": "https://stac-extensions.github.io/storage/v1.0.0/schema.json"
```

**Review by philvarner — COMMENTED** [review]
Few changes requested

File: examples/example-naip.json (hunk 1 above)
Comment (m-mohr) [reply examples/example-naip.json:2]: 1.0.0-rc.1?

**Comment by matthewhanson** [comment]
I'm approving and merging this.  We'll want to publish since it was still a WIP in the PR.  Think we still need a little discussion about it.

**Pull request merged**